notebooks/thunder_as_torch_compile_backend_tutorial.ipynb

```diff
@@ -0,0 +1,308 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #13.    print(opt_foo1(torch.randn(4, 4, requires_grad=True, device="cuda"), torch.randn(4, 4, requires_grad=True, device="cuda")))
</pre><p> </p><p>This is cool. I suggest creating a line to create the input tensor <code>a</code> , then running the function eagerly and getting <code>eager_output</code>, then running the ThunderFX version and getting <code>thunder_output</code>, then printing the <code>eager_output</code> and <code>thunder_output</code> and comparing them using <code>assert_close</code>,I think that will be more readable.</p><p> </p><p>This first example should probably use tensors on the cpu instead of a CUDA device. Then a second cell can state that Thunder supports CPU and CUDA tensors, and repeat the exercise with CUDA tensors. It would also be nice if the notebook detected if there wasn't a supported GPU and just printed that the CUDA part of the tutorial couldn't run because a supported GPU wasn't found. We should also note that while ThunderFX supports tensors on the CPU and a CUDA device, its optimizations currently target programs run with CUDA tensors.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Lightning-AI/lightning-thunder/pull/1524/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>